pkg/hubble/parser/threefour/parser.go

```diff
@@ -316,13 +318,25 @@ func (p *Parser) resolveEndpoint(ip net.IP, datapathSecurityIdentity uint32) *pb
 	if p.endpointGetter != nil {
 		if ep, ok := p.endpointGetter.GetEndpointInfo(ip); ok {
 			epIdentity := resolveIdentityConflict(ep.GetIdentity())
-			return &pb.Endpoint{
+			e := &pb.Endpoint{
 				ID:        uint32(ep.GetID()),
 				Identity:  epIdentity,
 				Namespace: ep.GetK8sNamespace(),
 				Labels:    sortAndFilterLabels(p.log, ep.GetLabels(), epIdentity),
 				PodName:   ep.GetK8sPodName(),
 			}
+			if p.pod == nil {
+				p.pod = ep.GetPod()
+			}
+			if p.pod != nil {
```

Comment:
So sorry. I should have been more clear. I meant something like this:
```golang
if pod := ep.GetPod; pod != nil {
	for _, owner := range pod.GetOwnerReferences() {
		e.WorkloadKinds = append(e.WorkloadKinds, owner.Kind)
                e.WorkloadNames = append(e.WorkloadNames, owner.Name)
        }
}
```

Reply:
Ah... Just updated.